app/javascript/components/public_health/CustomExport.js

```diff
@@ -0,0 +1,508 @@
+                    style={{ outline: 'none', boxShadow: 'none' }}
+                    onClick={() => this.handlePresetChange('config.format', 'xlsx')}>
+                    <FontAwesomeIcon className="mr-1" icon={['fas', 'file-excel']} />
+                    Excel
+                  </Button>
+                </Form.Group>
+              </Col>
+              <Col md={6} className="px-1 pt-2">
+                <Form.Group className="mb-0 float-right">
+                  {this.state.preset?.id ? (
+                    <React.Fragment>
+                      <Button
+                        id="custom-export-action-delete"
+                        size="sm"
+                        variant="danger"
+                        disabled={!this.state.preset?.id}
+                        className="mr-1"
+                        style={{ outline: 'none', boxShadow: 'none' }}
+                        onClick={this.delete}>
+                        <FontAwesomeIcon className="mr-1" icon={['fas', 'trash']} />
+                        Delete
+                      </Button>
+                      <Button
+                        id="custom-export-action-update"
+                        size="sm"
+                        variant="primary"
+                        disabled={!this.state.preset?.id}
+                        className="ml-1"
+                        style={{ outline: 'none', boxShadow: 'none' }}
+                        onClick={this.update}>
+                        <FontAwesomeIcon className="mr-1" icon={['fas', 'pen-alt']} />
+                        Update
+                      </Button>
+                    </React.Fragment>
+                  ) : this.state.preset?.name === '' ? (
+                    <OverlayTrigger overlay={<Tooltip>Please indicate a name for the saved Custom Export</Tooltip>}>
+                      <span>
+                        <Button size="sm" variant="primary" disabled style={{ outline: 'none', boxShadow: 'none', pointerEvents: 'none' }}>
+                          <FontAwesomeIcon className="mr-1" icon={['fas', 'save']} />
+                          Save
+                        </Button>
+                      </span>
+                    </OverlayTrigger>
+                  ) : (
+                    <Button id="custom-export-action-save" size="sm" variant="primary" style={{ outline: 'none', boxShadow: 'none' }} onClick={this.save}>
+                      <FontAwesomeIcon className="mr-1" icon={['fas', 'save']} />
+                      Save
+                    </Button>
+                  )}
+                </Form.Group>
+              </Col>
+            </Row>
+          </div>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button id="custom-export-action-cancel" variant="secondary btn-square" onClick={this.props.onClose}>
+            Cancel
+          </Button>
+          {this.state.preset?.config?.data?.patients?.checked?.length === 0 &&
```

Comment:
Might be nice to break this out into a helper like `can_save_export` that checks all this stuff.

Reply:
https://github.com/SaraAlert/SaraAlert/pull/562/commits/1f13143f8968feff3c7e9ada6139b47e5cd4ecff#diff-14793b3bc71143b45317a3d08474d370db9ffdfb9643e754341eeda1ab5f096bR166